Globals/GLOBAL.gd

```diff
@@ -64,3 +64,20 @@ func reloadDirectoryFiles(path, destinationArray):
 	else:
 		print_debug("Unable to open desired directory: " + path)
 		return !OK
+
+# From the ID present within the map file, find the appropriate unit, create and return that unit object
+func getUnitFromId(id):
+	var newUnit = null
+	for m in FACTIONCUSTODIAN.factionArray.size():
+		for n in FACTIONCUSTODIAN.factionArray[m].unitRoster.size():
+			if id == FACTIONCUSTODIAN.factionArray[m].unitRoster[n].id:
+				print_debug("found it!")
+				var unitTemplate = FACTIONCUSTODIAN.factionArray[m].unitRoster[n]
+				newUnit = FACTIONCUSTODIAN.factionArray[m].unitRoster[n]
+				newUnit = Unit.new(null)
+				newUnit.copyFrom(unitTemplate)
```

Comment:
the constructor can just take the unit template right?

Reply:
the constructor already takes a filepath for preexisting functions, and we can't create multiple constructors...